plugins/runner/internal/runtimehandler/binary/handler.go

```diff
@@ -0,0 +1,241 @@
+	h.pluginServerEndpoint = fmt.Sprintf("http://127.0.0.1:%d", port)
+
+	// Set environment variables
+	env := image.Metadata.Config.Config.Env
+	env = append(env, fmt.Sprintf("%s=%s", plugin.EnvListenAddress, h.pluginServerEndpoint))
+
+	// Set workdir
+	workDir := image.Metadata.Config.Config.WorkingDir
+	if workDir == "" {
+		workDir = "/"
+	}
+
+	// Initialize command
+	h.cmd = exec.CommandContext(ctx, args[0], args[1:]...) //nolint:gosec
+	h.cmd.Env = env
+	h.cmd.Dir = workDir
+
+	h.cmd.SysProcAttr = &syscall.SysProcAttr{
+		Chroot: h.pluginDir,
+	}
+
+	h.cmd.Stdin = &bytes.Buffer{}
+	h.stdoutPipe, err = h.cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("unable to open pipe for command stdout: %w", err)
+	}
+
+	h.stderrPipe, err = h.cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("unable to open pipe for command stderr: %w", err)
+	}
+
+	// Start command
+	openPortListener.Close()
+	err = h.cmd.Start()
+	if err != nil {
+		return fmt.Errorf("unable to start process: %w", err)
+	}
+
+	// Waiting for command to be finished in the background
+	go func() {
+		h.cmd.Wait() //nolint:errcheck
+	}()
+
+	h.ready = true
+
+	return nil
+}
+
+func (h *binaryRuntimeHandler) Ready() (bool, error) {
+	if h.cmd == nil {
+		return false, errors.New("plugin process is not running")
+	}
+
+	return h.ready, nil
+}
+
+func (h *binaryRuntimeHandler) GetPluginServerEndpoint(ctx context.Context) (string, error) {
+	return h.pluginServerEndpoint, nil
+}
+
+func (h *binaryRuntimeHandler) Logs(ctx context.Context) (io.ReadCloser, error) {
+	if h.cmd == nil {
+		return nil, errors.New("plugin process is not running")
+	}
+
+	stdoutPipeReader, stdoutPipeWriter := io.Pipe()
+	stderrPipeReader, stderrPipeWriter := io.Pipe()
+
+	go func() {
+		io.Copy(stdoutPipeWriter, h.stdoutPipe) //nolint:errcheck
+	}()
+
+	go func() {
+		io.Copy(stderrPipeWriter, h.stderrPipe) //nolint:errcheck
+	}()
+
+	return io.NopCloser(io.MultiReader(stdoutPipeReader, stderrPipeReader)), nil
+}
+
+func (h *binaryRuntimeHandler) Result(ctx context.Context) (io.ReadCloser, error) {
+	f, err := os.Open(filepath.Join(h.pluginDir, runtimehandler.RemoteScanResultFileOverride))
+	if err != nil {
+		return nil, fmt.Errorf("unable to open result file: %w", err)
+	}
+
+	return f, nil
+}
+
+func (h *binaryRuntimeHandler) Remove(ctx context.Context) error {
+	var removeErr error
+
+	if h.cmd.ProcessState != nil {
+		if !h.cmd.ProcessState.Exited() {
+			if err := h.cmd.Process.Kill(); err != nil {
+				removeErr = multierror.Append(removeErr, fmt.Errorf("failed to kill plugin process: %w", err))
+			}
+		}
+	}
+
+	// Unmount input directory
+	if err := syscall.Unmount(h.inputDirMountPoint, 0); err != nil {
+		removeErr = multierror.Append(removeErr, fmt.Errorf("failed to kill plugin process: %w", err))
+	} else {
+		// Call the cleanup function for the image only after the input directory is unmounted, or else it will also remove
+		// the root filesystem mounted under input
+		h.imageCleanup()
```

Comment:
in case we want to preserve the image between runs, would this have any effect on the rest of the flow when binary plugin scan is re-executed for the same input?

Reply:
also, is there a possibility that the root filesystem will ever be deleted with the existing approach?

Reply:
I also have my concerns, if it is okay not to clean up the plugin scanner image from the machine after the scan, we could drop the image cleanup altogether from the binary plugin handler.

Also during implementation, we optimized for the fact that within VMClarity, one scan will only have one input, but in the light of recent developments, we should rethink some parts of the code.

Reply:
I think there are multiple things here that we should think it through.

1. preserving the image

If the goal is to have a local cache of the image and save execution time/bandwidth by not re-downloading the image every time, then we can store the tar file somewhere else, and extract it on a plugin execution.

With the current approach we are mixing this local cache functionality with the plugin execution environment (the `/home/ubuntu/.vmclarity/plugins/kics` directory is the image cache and the root filesystem of the chroot environment where the kics plugin runs), that has multiple potential consequences:
- at some point we need to invalidate the cache, check if there is a new version, which can be a source of race condition if there is a running plugin in the directory
- it's not safe to run concurrently (currently the plugin family runner executes the plugins one after the other)

To solve this we can have a separate plugin cache (the downloaded tar file of the images), and on plugin execution we can extract the image tar file from our local cache, and create a dedicated and unique directory that we clean up after one plugin execution.

2. accidental deletion of root filesystem

With the read-only bind mount this is not supposed to happen in any case, yet somehow it does. I have a feeling that something is off here and we should investigate this further, understand what is happening in the background, so we can prevent from happening in the future.

We have other options to solve this issue, but that would complicate the implementation, and we would getting closer to a full-blown container runtime implementation.

Reply:
@akijakya will create an issue to summarise expected changes after the scanner refactor ticket and we will track all these expected changes there. it is not relevant to address it now